### PR TITLE
Inspect extensions to get source information

### PR DESCRIPTION
**Pull request by Syer10**
Continues #6410

This is an rewritten version of the Tachidesk Inspector that ArMor PR'd back in April. It takes the suggestions of other contributors, such as:

- Puts the array of source information(such as name, lang, and id) inside the index.json as a item in the extensions object. 
- Deletes all Tachidesk specific code, only adding whats needed for the Inspector
- Puts the Inspector.jar in the runner-files so that it is not downloaded externally

Sample from index.json
```json
[
  {
    "name": "Tachiyomi: Cubari",
    "pkg": "eu.kanade.tachiyomi.extension.all.cubari",
    "apk": "tachiyomi-all.cubari-v1.2.9.apk",
    "lang": "all",
    "code": 9,
    "version": "1.2.9",
    "nsfw": 0,
    "sources": [
      {
        "name": "Cubari",
        "lang": "en",
        "id": 6338219619148106000
      },
      {
        "name": "Cubari",
        "lang": "all",
        "id": 1470847599087460400
      },
      {
        "name": "Cubari",
        "lang": "other",
        "id": 2013845246758512400
      }
    ]
  },
  ...
]
```

Successful build of the index.json:
https://github.com/Syer10/tachiyomi-extensions/blob/repo/index.json

Successful run for the Inspector.jar:
https://github.com/Syer10/tachiyomi-extensions/actions/runs/1061635114
The Inspector.jar only takes around 30 seconds to go over around 1000 extensions.

Repo for the Inspector.jar:
https://github.com/Syer10/TachiExtension-Inspector
This can be transferred to the organization

### CODE REVIEW

File: .github/workflows/build_push.yml [hunk 1]
```diff
@@ -128,6 +128,11 @@ jobs:
           ref: master
           path: master
 
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
```
Comment (arkon) [inline .github/workflows/build_push.yml:134]: Does it work with JDK11?

File: .github/workflows/build_push.yml [hunk 2]
```diff
@@ -137,6 +142,11 @@ jobs:
             ${{ secrets.KEY_STORE_PASSWORD }} \
             ${{ secrets.KEY_PASSWORD }}
 
+      - name: Run inspector
+        run: |
+          cd master
+          java -jar ./.github/runner-files/Inspector.jar "apk" "output.json" "tmp"
```
Comment (arkon) [inline .github/workflows/build_push.yml:148]: It'd be ideal if the JAR came from a trusted place (a repo release artifact, maybe) from a tachiyomiorg repo.

**Review by arkon — COMMENTED** [review]
Seems okay overall. I think I'd want the inspector stuff to be transferred over first though, so there's a vaguely higher trustworthiness for it.

For the future: I wonder if it'd also be useful for the inspector to extract our the base URLs? Could maybe even infer its NSFW-ness based on the annotations.

File: .github/workflows/build_push.yml (hunk 1 above)
Comment (Syer10) [reply .github/workflows/build_push.yml:134]: Tachidesk works on any JDK after 8, so the inspector should be the same

Comment (arkon) [reply .github/workflows/build_push.yml:134]: Might as well bump this from the get-go then.

**Comment by Syer10** [comment]
> For the future: I wonder if it'd also be useful for the inspector to extract our the base URLs? Could maybe even infer its NSFW-ness based on the annotations.

I can see this being useful for the repo actions and website, as well as the nsfw being useful in the app. It would be easy to add to the inspector.

File: .github/workflows/build_push.yml [hunk 3]
```diff
@@ -137,6 +142,13 @@ jobs:
             ${{ secrets.KEY_STORE_PASSWORD }} \
             ${{ secrets.KEY_PASSWORD }}
 
+      - name: Run inspector
+        run: |
+          cd master
+          INSPECTOR_LINK="$(curl -s "https://api.github.com/repos/tachiyomiorg/tachiyomi-extensions-inspector/releases/latest" | grep -o "https.*download\/[a-zA-Z0-9.\/-]*")"
```
Comment (arkon) [inline .github/workflows/build_push.yml:148]:
```
curl -s https://api.github.com/repos/tachiyomiorg/tachiyomi-extensions-inspector/releases/latest | jq '.assets[0].browser_download_url'
```

Comment (Syer10) [reply .github/workflows/build_push.yml]: Done

**Pull request merged**